core/src/components/tab-bar/test/basic/tab-bar.e2e.ts

```diff
@@ -0,0 +1,108 @@
+import { expect } from '@playwright/test';
+import { configs, test } from '@utils/test/playwright';
+
+configs().forEach(({ title, screenshot, config }) => {
+  test.describe(title('tab-bar: basic'), () => {
+    test('should not have visual regressions', async ({ page }) => {
+      await page.setContent(
+        `
+        <ion-tab-bar selected-tab="1">
+          <ion-tab-button tab="1">
+            <ion-label>Recents</ion-label>
+          </ion-tab-button>
+
+          <ion-tab-button tab="2">
+            <ion-label>Favorites</ion-label>
+            <ion-badge>23</ion-badge>
+          </ion-tab-button>
+
+          <ion-tab-button tab="3">
+            <ion-label>Settings</ion-label>
+          </ion-tab-button>
+        </ion-tab-bar>
+      `,
+        config
+      );
+
+      const tabBar = page.locator('ion-tab-bar');
+      await expect(tabBar).toHaveScreenshot(screenshot(`tab-bar-diff`));
+    });
+  });
+});
+
+/**
+ * This behavior needs to be tested in both modes and directions to
+ * make sure the safe area padding is applied only to that side
+ * regardless of direction
+ */
+configs().forEach(({ title, screenshot, config }) => {
+  test.describe(title('tab-bar: basic'), () => {
+    test.describe('safe area left', () => {
+      test('should have padding on the left only', async ({ page }) => {
```

Comment:
Why do we have two different tests? I figured we could have one test with an uneven amount of safe area  (i.e 20px on the left 40px on the right) to make sure that the safe area is being added to the correct side

Reply:
I thought about separating them but I felt it was easier to catch regressions and visually see that there was _no padding_ on the one side.

Reply:
I'm not sure I follow. If the we had 1 test that applied different amounts of left/right safe area we'd still see diffs. Reason why I'm asking about this is we effectively have 2 tests testing the same underlying functionality.

For example, if we applied 40px of left safe area padding and 20px of safe area padding we'd get the following screenshots and diff if this fix regresses in the future. 

(RTL direction, iOS mode, Chrome)

Base Screenshot | Regressed Screenshot | Diff |
| - | - | - |
| ![tab-bar-safe-area-left-ios-rtl-Mobile-Chrome-darwin](https://github.com/ionic-team/ionic-framework/assets/2721089/0295e9c8-d12c-4814-83d0-d0eae7bee9c7) | ![tab-bar-safe-area-left-ios-rtl-actual](https://github.com/ionic-team/ionic-framework/assets/2721089/690e6e2f-9ba0-430e-a3e9-477b4533509e) | ![tab-bar-safe-area-left-ios-rtl-diff](https://github.com/ionic-team/ionic-framework/assets/2721089/628a0d94-9816-48c1-b404-07360cf35905) |

This test applied the following CSS:

```css
 :root {
  --ion-safe-area-left: 40px;
  --ion-safe-area-right: 20px;
}
```

Reply:
Yeah I was just thinking it would be easier to tell if a change happened to the padding on the component itself but I suppose you would have to verify the diff either way. I can update it.